--- /tmp/tmp.2S5MPfCRGI/lua/kinv/shared/sh_0_kinventory.lua	2025-12-01 09:58:12.000000000 +0000
+++ /home/cookie/gmod/serverfiles/garrysmod/addons/0020_ps2-master/lua/kinv/shared/sh_0_kinventory.lua	2026-08-18 16:45:35.707543537 +0000
@@ -6,7 +6,7 @@
 function KInventory.RegisterItemClassMixin( className, mixin )
 	-- Class already loaded, apply mixin now
 	if KInventory.Items and KInventory.Items[className] then
-		print(KInventory.Items[className])
+		-- change by cookie9216
 		KInventory.ApplyMixin( KInventory.Items[className], mixin )
 	end
 
