--- /tmp/tmp.2S5MPfCRGI/lua/ps2/client/tabs/management_tab/settings/cl_dsettingssection.lua	2025-12-01 09:58:12.000000000 +0000
+++ /home/cookie/gmod/serverfiles/garrysmod/addons/0020_ps2-master/lua/ps2/client/tabs/management_tab/settings/cl_dsettingssection.lua	2026-08-18 16:45:35.717544484 +0000
@@ -161,7 +161,8 @@
 	
 	local settingPanel = self[creatorFn]( self, settingsPath, settingInfo )
 	self:AddSettingPanel( settingPanel )
-	if settingInfo.value then
+	-- change by cookie9216
+	if settingInfo.value ~= nil then
 		settingPanel:SetValue( settingInfo.value )
 	end
 	return settingPanel
